scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -99,13 +104,22 @@ object Reindex extends EsScript {
 
         Thread.sleep(1000)
 
-        reindexScroll(query.execute.actionGet)
+        if(fromTime.isEmpty){
+          reindexScroll(query.setQuery(matchAllQuery).execute.actionGet)
+        } else {
+          val recentChangesQuery = rangeQuery("lastModified").from(fromTime.get).to(DateTime.now)
+          reindexScroll(query.setQuery(recentChangesQuery).execute.actionGet)
+        }
 
         // TODO: deleteIndex when we are confident
         client.close()
       }
     }
-    EsClient.reindex
+    if(extraArgs.isEmpty) {
```

Comment:
A statement in the form:

val date = if(extraArgs.isEmpty) None else Some((DateTime.parse(extraArgs(0))))
EsClient.reindex(date)

is certainly more accessible to non-native Scala developers or newcomers than foldLeft.